src/test/java/com/direwolf20/buildinggadgets/util/UniqueBlockState.java

```diff
@@ -0,0 +1,320 @@
+package com.direwolf20.buildinggadgets.util;
+
+import com.google.common.collect.ImmutableMap;
+import net.minecraft.block.Block;
+import net.minecraft.block.material.EnumPushReaction;
+import net.minecraft.block.material.MapColor;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.state.BlockFaceShape;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.EnumBlockRenderType;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.Mirror;
+import net.minecraft.util.Rotation;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+public class UniqueBlockState implements IBlockState {
+
+    public static final IBlockState AIR = createNew();
+
+    private static int lastId = -1;
+
+    public static UniqueBlockState createWithId(int id) {
+        return new UniqueBlockState(id);
+    }
+
+    public static UniqueBlockState createNew() {
+        return new UniqueBlockState(++lastId);
+    }
+
+    private final int id;
+
+    private UniqueBlockState(int id) {
+        this.id = id;
+    }
+
+    @Override
+    public String toString() {
+        return "UniqueBlockState{" + id + "}";
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        UniqueBlockState that = (UniqueBlockState) o;
+        return id == that.id;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    //All of these methods should not be used
```

Comment:
Why implement IBlockAccess if you never use its methods?

Reply:
Testing the placement logic only need to access blockstates, and should only need blockstates even if we support tile entities -- you probably don't want consider two tile entities are the same only if their serialized form are the same.

Also do you meant `RegionBlockView`? This is the fake blockstate class.

Reply:
Not the best naming but if that's it's use then that's fine.